src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs

```diff
@@ -297,6 +281,16 @@ private async Task ProcessQueueAsync()
         }
     }
 
+    /// <summary>
+    /// Choose the best method handler for the given request.
+    /// </summary>
+    protected virtual IMethodHandler GetHandlerForRequest(IQueueItem<TRequestContext> work)
+    {
+        var handlers = _handlerProvider.GetMethodHandlers(work.MethodName, work.RequestType, work.ResponseType);
+
+        return handlers.First(h => string.IsNullOrEmpty(h.Metadata)).Value;
+    }
```

Comment:
do we really want thsi default behavior?  why not have all subtypes specify what behavior they want.

Reply:
Simplified default behavior to just call IHandlerProvider